packages/salesforcedx-vscode-apex-replay-debugger/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,56 @@
+import vscode = require('vscode');
+import TelemetryReporter from 'vscode-extension-telemetry';
+
+export class TelemetryService {
+  private static instance: TelemetryService;
+  private context: vscode.ExtensionContext | undefined;
+  private reporter: TelemetryReporter | undefined;
+
+  public static getInstance() {
+    if (!TelemetryService.instance) {
+      TelemetryService.instance = new TelemetryService();
+    }
+    return TelemetryService.instance;
+  }
+
+  public initializeService(
+    context: vscode.ExtensionContext,
+    isTelemetryEnabled: boolean
+  ): TelemetryReporter | undefined {
+    this.context = context;
+
+    // TelemetryReporter is not initialized if user has disabled telemetry setting.
+    if (this.reporter === undefined && isTelemetryEnabled) {
+      const extensionPackage = require(this.context.asAbsolutePath(
+        './package.json'
+      ));
+
+      this.reporter = new TelemetryReporter(
```

Comment:
Initializing TelemetryReporter this way since that's how other Microsoft extensions do it (e.g. https://github.com/Microsoft/vscode-azure-account/blob/bb54795b948a4cdb111dba46f677d592aa673687/src/telemetry.ts)